src/lib.rs

```diff
@@ -894,7 +894,7 @@ pub fn handle_key_subcommand(
 
             if let Secret(secret_key, _, _) = xprv_desc_key {
                 let desc_pubkey = secret_key.as_public(&secp).unwrap();
-                Ok(json!({"xpub": desc_pubkey.to_string()}))
+                Ok(json!({"xpub": desc_pubkey.to_string(), "xprv": secret_key.to_string()}))
```

Comment:
`secret_key.to_string()` here gives the parent xprv provided.

I was able to calculate the child xprv using the following
```
 let _child = xprv.derive_priv(&secp, &deriv_path).unwrap();
```
However, this does not contain the path for the descriptor and requires some additional sting concatenation. Is there a better way of going about doing this?
I looked into the DescriptorKey and DescriptorSecretKey Enums, was unable to find a way.

Reply:
I think what we need to do is below. It doesn't split the derivation path into hardended/unhardened parts but I think this is probably better behavior since it gives the user more control. The `test_key_derive` test will also need to be updated with this change. 

```rust
            let deriv_path: DerivationPath = DerivationPath::from_str(path.as_str())?;

            let derived_xprv = &xprv.derive_priv(&secp, &deriv_path)?;
            let origin:KeySource = (xprv.fingerprint(&secp), deriv_path);
            
            let derived_xprv_desc_key: DescriptorKey<Segwitv0> =
                 derived_xprv.into_descriptor_key(Some(origin), DerivationPath::default())?;
            
            if let Secret(desc_seckey, _, _) = derived_xprv_desc_key {
                let desc_pubkey = desc_seckey.as_public(&secp).unwrap();
                Ok(json!({"xpub": desc_pubkey.to_string(), "xprv": desc_seckey.to_string()}))
            } else {
                Err(Error::Key(Message("Invalid key variant".to_string())))
            }
```

Reply:
yeah, that looks more clear with the child xprv being explicitly derived at the path.